paywall/src/data-iframe/blockchainHandler/KeyStore.ts

```diff
@@ -0,0 +1,38 @@
+import { EventEmitter } from 'events'
+import { KeyResult } from './blockChainTypes'
+
+/**
+ * KeyStore holds the keys corresponding to the locks on the
+ * page. When it has been filled up (i.e., `addKey()` has been called
+ * `numLocks` times), it emits an event containing all the currently
+ * valid keys.
+ */
+export default class KeyStore extends EventEmitter {
+  keys: { [lockAddress: string]: KeyResult | null }
+
+  constructor(lockAddresses: string[]) {
+    super()
+    this.keys = {}
+    lockAddresses.forEach(lockAddress => {
```

Comment:
Oh that's rigth! strings! Sorry I overlooked that!

Reply:
No worries at all!